### PR TITLE
NDRS-1117: Backport handshake

### PR DESCRIPTION
**Pull request by marc-casperlabs**
This backports two 1.1.0 changes to `dev`, with the necessary alterations: https://github.com/CasperLabs/casper-node/pull/1256 (making the maximum message size depend on the chainspec instead of being hardcoded) https://github.com/CasperLabs/casper-node/pull/1263 (include the chainspec version in the handshake)

It also introduces the first set of rountrip tests that are ensuring we can serialize/deserialize network messages from 1.0.0 and onwards.

### CODE REVIEW

File: node/src/components/small_network.rs [hunk 1]
```diff
@@ -502,11 +507,9 @@ where
 
                 info!(our_id=%self.our_id, %peer_id, %peer_address, "established incoming connection");
                 // The sink is only used to send a single handshake message, then dropped.
-                let (mut sink, stream) = framed::<P>(transport).split();
-                let handshake = Message::Handshake {
-                    network_name: self.network_name.clone(),
-                    public_address: self.public_address,
-                };
+                let (mut sink, stream) =
+                    framed::<P>(transport, self.chain_info.maximum_net_message_size).split();
```
Comment (dwerner) [inline node/src/components/small_network.rs:511]: IMO it's worth mentioning this in the PR description - that we're allowing configuration of the max framed size for messages.

**Review by dwerner — APPROVED** [review]
lgtm, but `maximum_net_message_size` should be mentioned in the description (admittedly it's a backport)

File: node/src/components/small_network.rs (hunk 1 above)
Comment (marc-casperlabs) [reply node/src/components/small_network.rs:511]: Hmm, technically it's not configuration, but a chain spec value? It's mentioned in the linked PR https://github.com/CasperLabs/casper-node/pull/1256

Comment (marc-casperlabs) [reply node/src/components/small_network.rs:511]: I'll add it though.

**Pull request merged**